### PR TITLE
Add getCurrentIpAddress action

### PR DESCRIPTION
**Pull request by bachya**
**Checks**

- [x] I've checked there are no linting errors.
- [x] I've checked the code is still at 100% test coverage.

**Added Actions (if relevant)**

- `Get Current IP Address`

**Are you happy to be listed as a contributor on [Shortcuts.fun](https://shortcuts.fun)?**

Yes

**Any other information / comments**

This is my first foray into Node and TypeScript – ever (after 10 years of Python). After I got my feet under me, your guides worked great.

Overall, I'm still getting used to the architecture of a Node module, how TypeScript is managed, etc., so it's entirely possible I haven't done things elegantly. Please send any feedback my way; I plan on contributing a lot! 😎

### CODE REVIEW

**Comment by joshfarrant** [comment]
Hey @bachya :wave: Great to have you as a contributor!

This is a good-looking PR, thanks!

To say this is your first foray into Node and Typescript this is excellent, and the code you've written is perfect. The only thing I'd ask would be to add two more tests to `getCurrentIpAddress.spec.ts`, something like:

`builds a getCurrentIpAddress action when an address is passed`

and

`builds a getCurrentIpAddress action when a type is passed`

The tests will pass in either an address or type to the `getCurrentIpAddress` function and then check that the output matches what we'd expect. You can essentially copy [this test in setTorch.spec.ts](https://github.com/joshfarrant/shortcuts-js/blob/master/__tests__/actions/setTorch.spec.ts#L21-L32), but just change the `setting` to `address` in the first test and `type` in the second. Other than that the PR is good to go!

Hope that makes sense. If you run into any issues let me know and I'd be happy to help :+1:

File: src/interfaces/WF/WFIPAddressType.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
```
Comment (joshfarrant) [inline src/interfaces/WF/WFIPAddressType.ts]: Could this file be renamed to `WFIPAddressTypeOption` please? Just to keep it consistent with the name of the type it defines.

File: src/interfaces/WF/WFIPAddressAddress.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
```
Comment (joshfarrant) [inline src/interfaces/WF/WFIPAddressAddress.ts]: Could this file be renamed to `WFIPAddressSourceOption` please? Just to keep it consistent with the name of the type it defines.

File: src/interfaces/WF/WFWorkflowActionParameters.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ import WFGetDictionaryValueType from './WFGetDictionaryValueType';
 import WFHTTPBodyType from './WFHTTPBodyType';
 import WFHTTPMethod from './WFHTTPMethod';
 import WFInputType from './WFInputType';
+import WFIPAddressSourceOption from './WFIPAddressAddress';
+import WFIPAddressTypeOption from './WFIPAddressType';
```
Comment (joshfarrant) [inline src/interfaces/WF/WFWorkflowActionParameters.ts]: You'll need to update these imports once you've renamed the two files above.

File: src/interfaces/WF/WFIPAddressType.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+ * @typedef {('IPv4'|'IPv6')} WFIPAddressTypeOption
```
Comment (joshfarrant) [inline src/interfaces/WF/WFIPAddressType.ts]: Could you remove this whole typedef comment please? This is my fault - I meant to remove all of these when I was changing how the docs are generated a few days ago, clearly I missed a couple! They're no longer needed now that we've switched to TypeDoc to generate the docs.

I'm going to remove all of the remaining `@typedef` comments I left in by mistake, so if you could just remove them from this PR then that'll save me a job 🙂

File: src/interfaces/WF/WFIPAddressAddress.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+ * @typedef {('External'|'Local')} WFIPAddressSourceOption
```
Comment (joshfarrant) [inline src/interfaces/WF/WFIPAddressAddress.ts]: Could you remove this `typedef` please? See my other comment for a bit more info 👍

File: src/actions/getCurrentIpAddress.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { withActionOutput } from '../utils';
+
+import WFIPAddressSourceOption from '../interfaces/WF/WFIPAddressAddress';
+import WFIPAddressTypeOption from '../interfaces/WF/WFIPAddressType';
```
Comment (joshfarrant) [inline src/actions/getCurrentIpAddress.ts]: These imports will need updating as per my comments below.

**Comment by xAlien95** [comment]
@joshfarrant, should `runScriptOverSSH` be renamed to `runScriptOverSsh`?

[Time ago I asked](https://github.com/joshfarrant/shortcuts-js/pull/8#discussion_r235121730) to use `SSH` instead of `Ssh` to follow how some JS functions/properties are named (e.g. `.innerHTML`), without noticing that in the CONTRIBUTING section is expressly written to use camelCase:

> The file (and action) you're creating should be the name of the action as it appears in the Shortcuts app, camelCased.

**Comment by joshfarrant** [comment]
@xAlien95 I suppose we need to formally come down on one side or the other, and update the CONTRIBUTING.md to reflect that.

Take a look at #26

**Comment by joshfarrant** [comment]
Looking good!

**Pull request merged**